gensim/utils.py

```diff
@@ -1263,3 +1269,45 @@ def _iter_windows(document, window_size, copy=False, ignore_below_size=True):
     else:
         for doc_window in doc_windows:
             yield doc_window.copy() if copy else doc_window
+
+
+def walk_with_depth(top, topdown=True, onerror=None, followlinks=False, depth=0):
```

Comment:
This is a very good point; I replaced this with a wrapper on `os.walk` that just deduces the depth in the manner you suggested.